controllers/workspace/devworkspace_controller.go

```diff
@@ -281,9 +281,12 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 	workspace.Spec.Template = *flattenedWorkspace
 
-	err = ssh.AddSshAgentPostStartEvent(&workspace.Spec.Template)
-	if err != nil {
-		return r.failWorkspace(workspace, "Failed to add ssh-agent post start event", metrics.ReasonWorkspaceEngineFailure, reqLogger, &reconcileStatus), nil
+	// Include to experimental features list because it is not clear how to handle post start events in containers without sh.
+	if *workspace.Config.EnableExperimentalFeatures {
```

Comment:
I believe this could result in a null pointer exception since `workspace.Config.EnableExperimentalFeatures` might be null. Instead we should check if it's null beforehand:
```go
if workspace.Config.EnableExperimentalFeatures != null && *workspace.Config.EnableExperimentalFeature
```